src/graph.jl

```diff
@@ -0,0 +1,349 @@
+    """
+    
+    edge_types = CSV.read(edge_type_defs)
+
+    synth_edges_df = []
+    
+    for edge_row in eachrow(edge_types)
+
+        src_row = @linq synth_vertex_df |> 
+            where(:v_type .== edge_row.src_type) |> 
+            select(:v_hash, :v_name, :v_type)
+        
+        dst_row = @linq synth_vertex_df |> 
+            where(:v_type .== edge_row.dst_type) |>
+            select(:v_hash, :v_name, :v_type)
+        
+        if size(src_row)[1] >=1 && size(dst_row)[1] >= 1
+            src_vhash = src_row.v_hash[1]
+            src_name = src_row.v_name[1]
+            src_vtype = src_row.v_type[1]
+
+            dst_vhash = dst_row.v_hash[1]
+            dst_name = dst_row.v_name[1]
+            dst_vtype = dst_row.v_type[1]
+            
+            if edge_row.value_field != "nothing"
+                edge_val = ("placeholder")
+            else
+                edge_val = nothing
+            end
+
+            synth_edge_attrs = NamedTuple{(:src_vhash, :src_name, :src_vtype, :dst_vhash, :dst_name, :dst_vtype, :edge_relation, :edge_description, :value)}(("$src_vhash", src_name, src_vtype, "$dst_vhash", dst_name, dst_vtype, edge_row.edge_relation, edge_row.description, edge_val))
+            push!(synth_edges_df, synth_edge_attrs)
+            
+        else
+            continue
+        end
+    end
+    
+    open(output_path, "w") do io
+        print(io, repr(load_graph_data(synth_edges_df)))
+    end
+
+    #CSV.write(output_path, synth_edges_df)
+    @info "Synthetic edge dataframe generated and saved as a Julia file."
+    
+    return DataFrame(synth_edges_df)
+
+end
+
+function insert_vertices_from_jl(vertices_file::String, input_graph::Union{MetaDiGraph, Nothing})
+    
+    # Ingest data frame containing vertex information; each row represents a vertex name and type
+    # vertex_id is computed upon insertion via a hash function, and is assumed to be unique 
+    # The tuple {vertex_name, vertex_type} is assumed to be unique 
+
+    v_df = DataFrame(include(vertices_file))
+    
+    # Case 1: the graph is currently empty; objective is to create it from the input vertex and edge files.
+    if isa(input_graph, Nothing)
+        
+        n_vertices = size(v_df,1)
+        G = MetaDiGraph(n_vertices)
+        counter = 1::Int64
+
+        set_indexing_prop!(G, :v_hash)
+        
+        for v in eachrow(v_df)
+        
+            v_hash = hash(string(v.v_name, v.v_type)) 
+            
+            v_attrs = Dict(
+                :v_name=>v.v_name,
+                :v_type=>v.v_type,
+                :used=>true)
+            
+            set_indexing_prop!(G, counter, :v_hash, "$v_hash")
+            set_props!(G, counter, v_attrs)
+            @info "Inserting new vertex into graph: $v"
+            counter += 1
+        end
+    
+    # Case 2: the graph already exists (as a graph object); objective is to add new vertices/edges, and increment weights when existing edges are encountered.
```

Comment:
This should definitely be broken into two methods. Multiple dispatch is ideal for splitting big functions like this up.

Reply:
👍